File: src/Services/ScraperService/src/Models/Event.ts
import { User } from "./User";

export interface Event {
    title: string;
    host: User;
    startDate?: Date;
    endDate?: Date;
    createdDate?: Date;
    lastUpdatedDate?: Date;
    isPrivate: boolean;
    adultsOnly: boolean;
    isPaid: boolean;
    maxNumberOfAttendees?: number;
    url: string;
    description?: string;
    location: Location;
    keywords: any[];
    attendees: string[];
    dateDisplay?: string;
    images?: string[];
    price?: string;
}

export interface Location {
    country: string;
    city: string;
    postalCode: string;
    streetName: string;
    streetNumber: string;
    housenumber?: string;
    floor?: string;
}

File: src/Services/EventManagementService/EventManagementService.Domain/Models/Events/Event.cs
namespace EventManagementService.Domain.Models.Events;

public class Event
{
    public int Id { get; set; }
    public string Title { get; set; }
    public DateTimeOffset StartDate { get; set; }
    public DateTimeOffset EndDate { get; set; }
    public DateTimeOffset CreatedDate { get; set; }
    public DateTimeOffset LastUpdateDate { get; set; }
    public bool IsPrivate { get; set; }
    public bool AdultsOnly { get; set; }
    public bool IsPaid { get; set; }
    public User Host { get; set; }
    public int MaxNumberOfAttendees { get; set; }
    public string Url { get; set; }
    public string? Description { get; set; }
    public string Location { get; set; }
    public string City { get; set; }
    public GeoLocation GeoLocation { get; set; }
    public string AccessCode { get; set; }
    public Category Category { get; set; }
    public IEnumerable<Keyword> Keywords { get; set; }
    public IEnumerable<string> Images { get; set; }
    public IEnumerable<User> Attendees { get; set; }
}

public class GeoLocation
{
    public float Lat { get; set; }
    public float Lng { get; set; }
}